### DISCUSSION TITLE
Validation rule for unicity of path+label

### DISCUSSION THREAD

**Discussion by BoD** [graphql/defer-stream-wg#37] [Q&A] [answered]
This is kind of a follow-up to #35. Would it make sense to have a rule to enforce unicity of path+label?

The spec currently says:
> When the response of the GraphQL operation is an event stream, the first value
will be the initial response. All subsequent values may contain `label` and
`path` entries. These two entries are used by clients to identify the `@defer`
or `@stream` directive from the GraphQL operation that triggered this value to
be returned by the event stream. **The combination of these two entries must be
unique across all values returned by the event stream.**

In certain cases, for instance, 

```graphql
query GetHero { 
  hero { 
    ...heroDetails1 @defer 
    ...heroDetails2 @defer 
  } 
}
```

it is guaranteed that path+label won't be unique and it could be detected.

I guess the question can be summarized as: 
- instead of having the recently added rule "Defer And Stream Directive Labels Are Unique" should there be a rule about the unicity of the combination of the path and label?

**Comment by robrichard** [comment, marked as answer]
This was written when label was still a mandatory argument. I think it should be updated now that we've agreed that it should be optional. It was meant to inform that neither the path nor label could be used to uniquely identify payloads, both are required. I think I will change it to something like 
> When a `label` is provided, the combination of these two entries will be unique across all values returned by the event stream.

**Reply by BoD** [reply]
Thanks for the reply and sorry for the late reply! 

What are your thoughts on making the unicity of path+label on any `@defer` directive an explicit validation rule though?

**Reply by robrichard** [reply]
The general sentiment around the label field is that it's necessary for most clients for now, but may become unnecessary in the future with proposals like fragment modularity or `__fulfilled` that was discussed a few times. So I don't think we should have any validation rules that add requirements for labels. However, it's fine for client libraries to have their own validation requiring labels if they are needed.
